sensors/hall effect sensor/rpm/hall_effect_sensor.cpp

```diff
@@ -0,0 +1,83 @@
+//NOTE: CODE TAKEN FROM: https://www.youtube.com/watch?v=u2uJMJWsfsg
+
+#include "hall_effect_sensor.h"
+
+
+HallEffectSensor::HallEffectSensor(const int &pin)
+  : Sensor(&pin)
+{
+  attachInterrupt(digitalPinToInterrupt(2), Pulse_Event, RISING);  
+  delay(1000);
+}
+
+void HallEffectSensor::read_sensor_value()
+{
+  LastTimeCycleMeasure = LastTimeWeMeasured;  
+  CurrentMicros = micros();  
+
+  if(CurrentMicros < LastTimeCycleMeasure)
+  {
+    LastTimeCycleMeasure = CurrentMicros;
+  }
+
+  // Calculate the frequency:
+  FrequencyRaw = 10000000000 / PeriodAverage; 
```

Comment:
Use a macro/variable for the number? It's not really intuitive what it is and I always lose track counting zeros :')